src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -233,6 +234,66 @@ public OrganisationsDetailResponse retrieveOrganisations() {
         return updatedOrganisationDetails;
     }
 
+    @Override
+    public OrganisationsDetailResponse retrieveOrganisations(String group) {
+        List<Organisation> retrievedOrganisations = organisationRepository.findAll();
+
+        List<Organisation> pendingOrganisations = new ArrayList<>();// = organisationRepository.findByStatus(OrganisationStatus.PENDING.name());
+        List<Organisation> activeOrganisations = new ArrayList<>();// = retrieveActiveOrganisationDetails();
+        List<Organisation> resultingOrganisations = new ArrayList<>();
+
+        retrievedOrganisations.forEach(organisation -> {
+            if (organisation.getStatus() == OrganisationStatus.ACTIVE) {
+                activeOrganisations.add(organisation);
+            }
+        });
+        retrievedOrganisations.forEach(organisation -> {
+            if (organisation.getStatus() == OrganisationStatus.PENDING) {
+                pendingOrganisations.add(organisation);
+            }
+        });
+
+        if (pendingOrganisations.isEmpty() && activeOrganisations.isEmpty()) {
+
+            log.info("No Organisations Retrieved...");
+            throw new EmptyResultDataAccessException(1);
+        }
+
+        List<Organisation> updatedOrganisationDetails = new ArrayList<>();
+        Map<String, Organisation> activeOrganisationDtls = new ConcurrentHashMap<>();
+
+        activeOrganisations.forEach(organisation -> {
+            if (!organisation.getUsers().isEmpty() && null != organisation.getUsers().get(0).getUserIdentifier()) {
```

Comment:
if the intention is to create a map of Active orgs then you can perhaps create it in the first loop itself instead of creating a list first and then converting it to map

Reply:
i was just preserving earlier functionality and made my code work around this but i will look into it

Reply:
is this refactored?

Reply:
yes, this file is outdated